src/preload/preload.js

```diff
@@ -1,141 +1,386 @@
-// This adds the "preload" extension to htmx.  By default, this will
-// preload the targets of any tags with `href` or `hx-get` attributes
-// if they also have a `preload` attribute as well.  See documentation
-// for more details
-htmx.defineExtension('preload', {
+/**
+ * This adds the "preload" extension to htmx. The extension will 
+  * preload the targets of elements with "preload" attribute if:
+  * - they also have `href`, `hx-get` or `data-hx-get` attributes
+  * - they are radio buttons, checkboxes, select elements and submit
+  *   buttons of forms with `method="get"` or `hx-get` attributes
+  * The extension relies on browser cache and for it to work
+  * server response must include `Cache-Control` header
+  * e.g. `Cache-Control: private, max-age=60`.
+  * For more details @see https://htmx.org/extensions/preload/
+ */
 
+htmx.defineExtension('preload', {
   onEvent: function(name, event) {
-    // Only take actions on "htmx:afterProcessNode"
-    if (name !== 'htmx:afterProcessNode') {
+    // Process preload attributes on `htmx:afterProcessNode`
+    if (name === 'htmx:afterProcessNode') {
+      // Initialize all nodes with `preload` attribute
+      const parent = event.target || event.detail.elt;
+      const preloadNodes = [
+        ...parent.hasAttribute("preload") ? [parent] : [],
+        ...parent.querySelectorAll("[preload]")]
+      preloadNodes.forEach(function(node) {
+        // Initialize the node with the `preload` attribute
+        init(node)
+
+        // Initialize all child elements which has
+        // `href`, `hx-get` or `data-hx-get` attributes
+        node.querySelectorAll('[href],[hx-get],[data-hx-get]').forEach(init)
+      })
       return
     }
 
-    // SOME HELPER FUNCTIONS WE'LL NEED ALONG THE WAY
+    // Intercept HTMX preload requests on `htmx:beforeRequest` and
+    // send them as XHR requests instead to avoid side-effects,
+    // such as showing loading indicators while preloading data. 
+    if (name === 'htmx:beforeRequest') {
+      const requestHeaders = event.detail.requestConfig.headers
+      if (!("HX-Preload" in requestHeaders
+            && requestHeaders["HX-Preload"] === "true")) {
```

Comment:
For continuity, I believe this should be HX-Preloaded instead of HX-Preload, similar to how adding hx-boost will add the header HX-Boosted
https://htmx.org/reference/#request_headers

Reply:
@ALameLlama Nicholas I agree. I have renamed it.